src/components/application_manager/test/commands/mobile/set_app_icon_test.cc

```diff
@@ -144,6 +154,555 @@ TEST_F(SetAppIconRequestTest, OnEvent_UI_UNSUPPORTED_RESOURCE) {
+  const std::string actual_sync_file_name =
+      (*result)[am::strings::msg_params][am::strings::sync_file_name]
+               [am::strings::value].asString();
+  EXPECT_EQ(expect_sync_file_name, actual_sync_file_name);
+
+  const uint32_t expect_app_id = app_id;
+  const uint32_t actual_app_id = static_cast<uint32_t>(
+      (*result)[am::strings::msg_params][am::strings::app_id].asInt());
+  EXPECT_EQ(expect_app_id, actual_app_id);
+}
+
+TEST_F(SetAppIconRequestTest, Run_NonExistentApp_ApplicationNotRegistered) {
+  MessageSharedPtr msg = CreateFullParamsUISO();
+  const std::string sync_file_name = "sync_file";
+  (*msg)[am::strings::msg_params][am::strings::sync_file_name] = sync_file_name;
+
+  utils::SharedPtr<SetAppIconRequest> req =
+      CreateCommand<SetAppIconRequest>(msg);
+
+  MockAppPtr mock_app;
+  EXPECT_CALL(app_mngr_, application(kConnectionKey))
+      .Times(1)
+      .WillOnce(Return(mock_app));
+
+  MessageSharedPtr result;
+  EXPECT_CALL(app_mngr_, ManageMobileCommand(_, _))
+      .WillOnce(DoAll(SaveArg<0>(&result), Return(true)));
+
+  req->Run();
+
+  const mobile_apis::Result::eType expect_result_code =
+      mobile_apis::Result::APPLICATION_NOT_REGISTERED;
+  const mobile_apis::Result::eType actual_result_code =
+      static_cast<const mobile_apis::Result::eType>(
+          (*result)[am::strings::msg_params][am::strings::result_code].asInt());
+  EXPECT_EQ(expect_result_code, actual_result_code);
+}
+
+TEST_F(SetAppIconRequestTest,
+       Run_SyncFileNameWithForbiddenSymbols_InvalidData) {
+  MessageSharedPtr msg = CreateFullParamsUISO();
+  const std::string sync_file_name = "sync_file/";
+  (*msg)[am::strings::msg_params][am::strings::sync_file_name] = sync_file_name;
+
+  utils::SharedPtr<SetAppIconRequest> req =
+      CreateCommand<SetAppIconRequest>(msg);
+
+  MockAppPtr mock_app = CreateMockApp();
+  EXPECT_CALL(app_mngr_, application(kConnectionKey))
+      .Times(1)
+      .WillOnce(Return(mock_app));
+
+  MessageSharedPtr result;
+  EXPECT_CALL(app_mngr_, ManageMobileCommand(_, _))
+      .WillOnce(DoAll(SaveArg<0>(&result), Return(true)));
+
+  req->Run();
+
+  const mobile_apis::Result::eType expect_result_code =
+      mobile_apis::Result::INVALID_DATA;
+  const mobile_apis::Result::eType actual_result_code =
+      static_cast<const mobile_apis::Result::eType>(
+          (*result)[am::strings::msg_params][am::strings::result_code].asInt());
+  EXPECT_EQ(expect_result_code, actual_result_code);
+}
+
+TEST_F(SetAppIconRequestTest, Run_NonExistentFile_InvalidData) {
+  MessageSharedPtr msg = CreateFullParamsUISO();
+  const std::string sync_file_name = "sync_file";
+  (*msg)[am::strings::msg_params][am::strings::sync_file_name] = sync_file_name;
+
+  utils::SharedPtr<SetAppIconRequest> req =
+      CreateCommand<SetAppIconRequest>(msg);
+
+  MockAppPtr mock_app = CreateMockApp();
+  EXPECT_CALL(app_mngr_, application(kConnectionKey))
+      .Times(1)
```

Comment:
@BSolonenko redundant
Read [GMOCK](https://github.com/google/googletest/blob/master/googlemock/docs/ForDummies.md#cardinalities-how-many-times-will-it-be-called)

Reply:
@AByzhynar Removed in https://github.com/smartdevicelink/sdl_core/pull/1974/commits/d3e063fe7994d5ed83665c9492a000bb0dc47552